src/main/scala/beam/agentsim/infrastructure/ParkingFunctions.scala

```diff
@@ -153,7 +153,9 @@ class ParkingFunctions[GEO: GeoLevel](
     parkingZone: ParkingZone[GEO],
     geoArea: GEO
   ): Coord = {
-    if (parkingZone.reservedFor.managerType == VehicleManager.TypeEnum.Household)
+    if (parkingZone.link.isDefined)
+      parkingZone.link.get.getCoord
+    else if (parkingZone.reservedFor.managerType == VehicleManager.TypeEnum.Household)
```

Comment:
Generally GeoLvel.geoSampling for Link type returns the link coord here. Could we just use parking files on the link level?

Reply:
as I explained above, what I tried to do is to make a hybrid form, where in TAZ GEO, some stalls are at link level. Do you have a better approach ?